tuta-sdk/rust/sdk/src/services/service_executor.rs

```diff
@@ -0,0 +1,650 @@
+#[cfg_attr(test, mockall_double::double)]
+use crate::crypto::crypto_facade::CryptoFacade;
+use crate::entities::entity_facade::EntityFacade;
+use crate::entities::Entity;
+use crate::instance_mapper::InstanceMapper;
+use crate::json_element::RawEntity;
+use crate::json_serializer::JsonSerializer;
+use crate::metamodel::TypeModel;
+use crate::rest_client::{HttpMethod, RestClient, RestClientOptions};
+use crate::rest_error::HttpError;
+use crate::services::hidden::Executor;
+use crate::services::{
+	DeleteService, ExtraServiceParams, GetService, PostService, PutService, Service,
+};
+use crate::type_model_provider::TypeModelProvider;
+use crate::{ApiCallError, HeadersProvider};
+use serde::Deserialize;
+use serde::Serialize;
+use std::sync::Arc;
+
+pub struct ServiceExecutor {
+	auth_headers_provider: Arc<HeadersProvider>,
+	crypto_facade: Arc<CryptoFacade>,
```

Comment:
I think this `CryptoFacade` here is the reason why we don't have strong guarantees about logged in state.

I think we should have similar separation as with `EntityClient` and have one without encryption support and one with encryption support.